pkg/skaffold/initializer/init.go

```diff
@@ -126,14 +134,43 @@ func DoInit(ctx context.Context, out io.Writer, c Config) error {
 			}
 			pairs = append(pairs, newPairs...)
 		} else {
-			resolved, err := resolveBuilderImages(unresolvedBuilderConfigs, unresolvedImages, c.Force)
-			if err != nil {
+			resolved, unresolved, err := resolveBuilderImages(unresolvedBuilderConfigs, unresolvedImages, c.Force)
+			switch {
```

Comment:
this whole logic here makes DoInit, the hardest to test part of `skaffold init` 30 lines longer. 
It is a nested switch inside an else ... with more ifs and for loops in there. 
This is not readable code.

Reply:
as we discussed out of band, this PR was meant to open a discussion primarily on the UX/decision tree of the feature. I should have been more clear about that! I agree that this entire function is a mess and needs to be refactored/rewritten.